File: CODE/Canteen/Backend-LatestV1/src/main/java/com/SmartCanteen/Backend/Entities/MenuItem.java
//package com.SmartCanteen.Backend.Entities;
//
//import com.fasterxml.jackson.annotation.JsonBackReference;
//import jakarta.persistence.*;
//import lombok.Data;
//
//import java.math.BigDecimal;
//
//@Entity
//@Data
//@Table(name = "menu_items")
//public class MenuItem {
//
//    @Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
//    private Long id;
//
//    @Column(nullable = false)
//    private String name;
//
//
//    @Column(nullable = false)
//    private BigDecimal price;
//
//    @Column(nullable = false)
//    private Integer stock;
//
//    @Column
//    private String imagePath;
//
//    @Version
//    private Integer version;
//
//    @ManyToOne
//    @JoinColumn(name = "category_id")
//    @JsonBackReference
//    private FoodCategory category;
//}


package com.SmartCanteen.Backend.Entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;

@Entity
@Data
@Table(name = "menu_items")
public class MenuItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private BigDecimal price;

    // --- NEW: Cost price for profit calculation ---
    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal costPrice;

    @Column(nullable = false)
    private Integer stock;

    // --- Path for the food item's image ---
    @Column
    private String imagePath;

    @Version
    private Integer version;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    @JsonBackReference
    private FoodCategory category;
}

File: mobile_app/food_app2/lib/models/menu_item.dart
class MenuItem {
  final int id;
  final String name;
  final double price;
  final double costPrice;
  final String imagePath;
  final int categoryId;
  final String? categoryName;
  final int stock;

  MenuItem({
    required this.id, required this.name, required this.price,
    required this.costPrice, required this.imagePath, required this.categoryId,
    this.categoryName, required this.stock,
  });

  factory MenuItem.fromJson(Map<String, dynamic> json) => MenuItem(
    id: json['id'], name: json['name'],
    price: (json['price'] as num).toDouble(),
    costPrice: (json['costPrice'] as num?)?.toDouble() ?? 0.0,
    imagePath: json['imagePath'] ?? '', categoryId: json['categoryId'] ?? 0,
    categoryName: json['categoryName'], stock: json['stock'] ?? 0,
  );
}